### ISSUE TITLE
Carbonite\NxMap.lua:6295: attempt to perform arithmetic on a nil value

### ISSUE THREAD

**Issue by Thrumbar** [mikepauer/Carbonite#463]
Got this on quest "The Diamond King" and "The Archmage" upon going to Ulduar and Karazhan by the teleport provided in quest or upon entering it.  Both produce the error upon entering the Scenario.....

3x Carbonite\NxMap.lua:6295: attempt to perform arithmetic on a nil value
Carbonite\NxMap.lua:6295: in function `UpdateOverlay'
Carbonite\NxMap.lua:6129: in function `UpdateZones'
Carbonite\NxMap.lua:4407: in function `Update'
Carbonite\NxMap.lua:3639: in function <Carbonite\NxMap.lua:3556>

Locals:
self = <table> {
 InstMapWX2 = 2727.3407068359
 MiniBlks = 6
 IconScale = 3
 MenuIShowWorld = <table> {
 }
 GIconMenuITogInst = <table> {
 }
 InstMapWY2 = -4859.1584203451
 InstMapInfo = <table> {
 }
 InstMapId = 1076
 MapPosY = -4865.0103086766
 MMScalesC = <table> {
 }
 LOpts = <table> {
 }
 TextFStrs = <table> {
 }
 PlyrLastDir = 2.0000059748505
 Scrolling = false
 Menu = <table> {
 }
 EffScale = 0.63999998569489
 Zone = 0
 WorldAlpha = 0
 StartupShown = true
 ScaleDraw = 4
 BackgndAlphaFade = 0.4
 MoveLastX = 2724.8802212267
 PlyrDir = 2.0000059748505
 PlyrSpeedCalcTime = 105834.033
 MapPosYDraw = -4865.0103086766
 PlyrSpeed = 0
 MapScale = 0.39920159680639
 CurMapBG = false
 PadX = 0
 BGIncNum = 0
 ViewSavedData = <table> {
 }
 MapW = 356.80059814453
 MMGathererUpdateDelay = 1
 LClickTime = 0
 MouseEnabled = true
 ButAutoScaleOn = <table> {
 }
 MapH = 317.00012207031
 MMScales = <table> {
 }
 MapPosX = 2724.8802212267
 BackgndAlphaTarget = 0.4
 MapsDrawnOrder = <table> {
 }
 BGGrowBars = true
 ArrowScroll = 0.33
 H = 141
 Tick = 34894
 IconNIFrms = <table> {
 }
 LastDungeonLevel = 1
 BaseScale = 0.025
 ToolBar = <table> {
 }
 MoveLastY = -4865.0103086766
 MenuIShowHerb = <table> {
 }
 DotPartyScale = 1
 PlyrX = 2724.8802212267
 QuestAlpha = 0.3
 InstMapWX1 = 2723.4266443359
 CurWorldUpdateOverlayNum = 1
 Scale = 4
 Targets = <table> {
 }
 BackgndAlpha = 0.4
 ArrowPulse = 1
 MenuIShowMine = <table> {
 }
 MapId = 0
 MMOwn = false
 Size1 = 1.1718750261934
 ShowUnexplored = true
 InstLevelSet = 1
 Arch = <unnamed> {
 }
 PIconMenu = <table> {
 }
 InstMapWY1 = -4866.9865453451
 LocTipFrm = NxMapTip {
 }
 IconFrms = <table> {
 }
 StepTime = 0
 PlyrSpeedX = 6603.9807724716
 PlyrFrm = <unnamed> {
 }
 DebugAdjustScale = 0.1
 ScaleDrawW = 0.25
 IconStaticFrms = <table> {
 }
 UpdateTrackingDelay = 0
 IconNavScale = 1
 TrackDistYd = 5.1291067662588
 TrackName = "Stormwind, Elwynn"
 MiniFrms = <table> {
 }
 InCombat = false
 Win = <table> {
 }
 InstanceId = 1076
 RealScale = 4
 Data = <table> {
 }
 TrackDir = false
 WorldHotspotTipStr = "Eastern Kingdoms, |cff20ff20Stormwind City (City)
"
 MenuIShowTimber = <table> {
 }
 CurOverlays = false
 LocTipFStrs = <table> {
 }
 TrackPlyrs = <table> {
 }
 TrackETA = false
 TextFrm = <unnamed> {
 }
 TextScFrm = <unnamed> {
 }
 TrackPlayer = "Deepaxe"
 GIconMenuIFindNote = <table> {
 }
 BGIconMenu = <table> {
 }
 QuestWin = <unnamed> {
 }
 WorldHotspotsCity = <table> {
 }
 PlyrY = -4865.0103086766
 TargetNextUniqueId = 1
 Cont = 90
 Tracking = <table> {
 }
 Guide = <table> {
 }
 WorldHotspots = <table> {
 }
 PlyrRZY = 75.73601603508
 Level = 7
 W = 150
 MapsDrawnFade = <table> {
 }
 DotRaidScale = 1
 ContFrms = <table> {
 }
 CurWorldUpdateMapId = -1
 TileFrms = <table> {
 }
 ScaleW = 0.25
 UpdateTargetDelay = 0
 CurOpts = <table> {
 }
 MenuIPlyrFollow = <table> {
 }
 TitleH = 0
 PlyrSpeedY = 1937.2988415742
 MMOwnedFrms = <table> {
 }
 MMChkDelay = 32
 ArchAlpha = 0.3
 MapIndex = 1
 MapPosXDraw = 2724.8802212267
 PlyrRZX = 37.137293815613
 KillShow = false
 MenuIMonitorZone = <table> {
 }
 NeedWorldUpdate = true
 GIconMenu = <table> {
 }
 TransMenu = <table> {
 }
 Frm = <unnamed> {
 }
 DotZoneScale = 1
 MoveDir = -40.543463581304
 BackgndAlphaFull = 1
 DotPalScale = 1
}
mapId = 0
bright = 1
noUnexplored = nil
wzone = <table> {
 1 = 10
 2 = 0
 3 = 0
 4 = 0
 5 = 0
 Overlay = "barrens"
 Connections = <table> {
 }
 Cont = 0
 Zone = 0
}
txFolder = "barrens"
overlays = <table> {
 dreadmistpeak = "290,104,241,195"
 thornhill = "481,254,239,231"
 thestagnantoasis = "344,379,336,289"
 farwatchpost = "555,129,207,332"
 thesludgefen = "403,6,257,249"
 groldomfarm = "448,127,243,217"
 thedryhills = "116,57,283,270"
 themerchantcoast = "556,456,315,212"
 boulderlodemine = "511,7,278,209"
 thewailingcaverns = "152,318,377,325"
 thecrossroads = "362,275,233,193"
 ratchet = "547,379,219,175"
 morshanrampart = "258,6,261,216"
 theforgottenpools = "100,208,446,256"
}
unex = nil
bW = nil
bH = nil
txIndex = nil
txPixelW = nil
txFileW = nil
txPixelH = nil
txFileH = nil
path = "Worldmap\barrens\"
alpha

**Comment by Thrumbar** [comment]
[12:31:27]Dump: value=GetInstanceInfo()
[12:31:27][1]="Ulduar",
[12:31:27][2]="scenario",
[12:31:27][3]=12,
[12:31:27][4]="Normal Scenario",
[12:31:27][5]=1,
[12:31:27][6]=0,
[12:31:27][7]=false,

**Comment by JWMcQuire** [comment]
When I saw your error report today (8/17/2016), I logged into Greymane and ran two alts (with the latest Carbonite installed) through "The Diamond King" and "Hiding in the Stacks" and experienced no errors on either alt.

I then logged into Cenarius and ran my DH through the same quests and got the same/similar error as you did when using the portal to Ulduar. I don't have any other alts on other servers high enough to test this further, but I can confirm that I get the same/similar bug at least on one realm when porting to Ulduar from Dalaran as did yours.

Since the bug I got was similar to yours, I won't re-post it unless asked to.
